lightning/src/routing/network_graph.rs

```diff
@@ -139,13 +143,15 @@ macro_rules! secp_verify_sig {
 impl<C: Deref + Sync + Send, L: Deref + Sync + Send> RoutingMessageHandler for NetGraphMsgHandler<C, L> where C::Target: chain::Access, L::Target: Logger {
 	fn handle_node_announcement(&self, msg: &msgs::NodeAnnouncement) -> Result<bool, LightningError> {
 		self.network_graph.write().unwrap().update_node_from_announcement(msg, &self.secp_ctx)?;
-		Ok(msg.contents.excess_data.is_empty() && msg.contents.excess_address_data.is_empty())
+		Ok(msg.contents.excess_data.len() <=  MAX_EXCESS_BYTES_FOR_RELAY &&
+		   msg.contents.excess_address_data.len() <= MAX_EXCESS_BYTES_FOR_RELAY &&
+		   msg.contents.excess_data.len() + msg.contents.excess_address_data.len() <= MAX_EXCESS_BYTES_FOR_RELAY)
```

Comment:
Any reason for the above two checks when the additive branch exists?

Reply:
Its probably not possible to hit just because of the deserialization limits, but I like to make sure code is obviously overflow-safe :)

Reply:
Good to know, thanks!